core/src/main/java/org/mule/runtime/core/internal/el/dataweave/DataWeaveExpressionLanguageAdaptor.java

```diff
@@ -268,12 +267,8 @@ public ExpressionLanguageSessionAdaptor openSession(ComponentLocation componentL
 
       @Override
       public TypedValue<?> evaluate(String expression, DataType expectedOutputType) throws ExpressionRuntimeException {
-        String sanitized = sanitize(expression);
-        if (isPayloadExpression(sanitized)) {
-          return resolvePayload(event, context);
-        }
         try {
-          return session.evaluate(sanitized, expectedOutputType);
+          return session.evaluate(sanitize(expression), expectedOutputType);
```

Comment:
is not the `isPayloadExpression()` still needed?

Reply:
nope, for this case it was wrong, we need to evaluate the expression to match the expected type